enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,169 @@
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "Metal3ProvisioningController" or just "Metal3Controller" would be
+created within the operator scope.
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying baremetal
+server which would be physically connected to the provisioning network. This
+configuration is needed only for the underlying provisioning service (Ironic)
+and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on the
+baremetal server for provisioning. This is also a value that is useful just to the
+provisioning service. This value should not be in the DHCP range and should not
+be used in the provisioning network for any other purpose (should be a free IP
+address.) It is expected to be provided in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its CIDR. The
+ProvisioningIP and the ProvisioningDHCPRange are expected to be within this network.
+The value for this config is expected to be provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which baremetal
+hosts can be assigned their IP address. This range should be provided as two
+comma seperated IP addresses where the addresses represent a contiguous range of
+IPs that are available to be allocated. It is expected to be provided in the format :
+10.1.0.10, 10.1.0.100.
+
+### User Stories [optional]
+
+This enhancement will allow users to have their Barametal IPI deployments
+customizable to their specific hardware and network requirements.
+
+This also allows IPI baremetal deployments to take place without manual
+workarounds like creating a ConfigMap for the config (which is the
+current approach we are using to test baremetal deployments.) 
+
+The new config items would be set by the installer and will be used by
+the machine-api-operator (MAO) to generate more config items that are
+derivable from these basic parameters. Put together, these config
+parameters are passed in as environment variables to the various containers
+that make up a metal3 baremetal IPI deployment. 
+
+## Design Details
+
+### Test Plan
+
+The test plan should involve making sure the openshift/installer generates
+all configuration items within the BaremetalPlatformStatus when the platform
+type is Baremetal.
+
+MAO reads this configuration and uses these to derive additional configuration
+required to bring up a metal3 cluster. E2e testing should make sure that MAO
+is able to bring up a metal3 cluster using config from this new Metal3Controller
+CR which has operator scope.
```

Comment:
The test plan should include actually deploying some worker nodes, as just seeing the services come up isn't enough to completely validate the configuration is correct.

Reply:
Yes, updated test plan with more test scenarios.